src/rai/rai/node.py

```diff
@@ -484,19 +485,36 @@ async def agent_loop(self, goal_handle: ServerGoalHandle):
                 payload, {"recursion_limit": self.AGENT_RECURSION_LIMIT}
             ):
 
-                print(state.keys())
                 graph_node_name = list(state.keys())[0]
                 if graph_node_name == "reporter":
                     continue
 
                 msg = state[graph_node_name]["messages"][-1]
-
                 if isinstance(msg, HumanMultimodalMessage):
                     last_msg = msg.text
+                elif isinstance(msg, BaseMessage):
+                    if isinstance(msg.content, list):
+                        assert len(msg.content) == 1
+                        last_msg = msg.content[0].get("text", "")
+                    else:
+                        last_msg = msg.content
                 else:
-                    last_msg = msg.content
-
-                if len(str(last_msg)) > 0 and "Retrieved state: {}" not in last_msg:
+                    raise ValueError(f"Unexpected type of message: {type(msg)}")
+
+                last_msg = self.simple_llm.invoke(
+                    [
+                        SystemMessage(
+                            content=(
+                                "You are an experienced reporter deployed on a autonomous robot. "  # type: ignore
+                                "Your task is to summarize the message in a way that is easy for other agents to understand. "
+                                "Do not use markdown formatting. Keep it short and concise. If the message is empty, please return empty string."
```

Comment:
Won't it introduce much latency? This node already return this report: https://github.com/RobotecAI/rai/blob/0c06a450fb8c0463dbe198a3ae1378b790604714/src/rai/rai/agents/state_based.py#L43
I think we should change it instead of adding another summarizer here.

Reply:
This is solely for the feedback. I find unprocessed feedback non informative. I am open for suggestions

Reply:
I was thinking about keeping a different history for this purpose with processed feedback added between main history messages. This is of course the least cost efficient however this approach may elevate the quality of the feedback as it will understand context

Reply:
I agree that it's not very readable now. Ok, let's merge it, but please add `NOTE` in the code to improve, because we can do it with more classical way.